### PR TITLE
feat: add autoConfigEmulator connection option

### PR DESCRIPTION
**Pull request by olavloite**
Adds `autoConfigEmulator` connection option. When this option is set to true for a connection:
1. The connection will by default try to connect to `localhost:9010` (unless a specific host/port is set in the connection URL).
2. Plain text communication will be enabled.
3. Authentication will be disabled.
4. The instance and database in the connection string will automatically be created on the emulator if any of them do not yet exist. Any existing instance and/or database will remain untouched.

Towards https://github.com/googleapis/java-spanner-jdbc/issues/380

### CODE REVIEW

File: google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionOptions.java [hunk 1]
```diff
@@ -483,18 +496,28 @@ private ConnectionOptions(Builder builder) {
...
     QueryOptions.Builder queryOptionsBuilder = QueryOptions.newBuilder();
     queryOptionsBuilder.setOptimizerVersion(parseOptimizerVersion(this.uri));
     this.queryOptions = queryOptionsBuilder.build();
     this.returnCommitStats = parseReturnCommitStats(this.uri);
+    this.autoConfigEmulator = parseAutoConfigEmulator(this.uri);
+    this.usePlainText = this.autoConfigEmulator || parseUsePlainText(this.uri);
+
+    if (matcher.group(Builder.HOST_GROUP) == null) {
+      if (this.autoConfigEmulator) {
+        this.host = "http://localhost:9010";
```
Comment (thiagotnunes) [inline google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionOptions.java:509]: nit: extract constant for this value

File: google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionOptions.java [hunk 2]
```diff
@@ -483,18 +496,28 @@ private ConnectionOptions(Builder builder) {
...
-    this.usePlainText = parseUsePlainText(this.uri);
     this.userAgent = parseUserAgent(this.uri);
     QueryOptions.Builder queryOptionsBuilder = QueryOptions.newBuilder();
     queryOptionsBuilder.setOptimizerVersion(parseOptimizerVersion(this.uri));
     this.queryOptions = queryOptionsBuilder.build();
     this.returnCommitStats = parseReturnCommitStats(this.uri);
+    this.autoConfigEmulator = parseAutoConfigEmulator(this.uri);
+    this.usePlainText = this.autoConfigEmulator || parseUsePlainText(this.uri);
+
+    if (matcher.group(Builder.HOST_GROUP) == null) {
```
Comment (thiagotnunes) [inline google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionOptions.java:507]: nit: extract method for the `if` `else` block

**Review by thiagotnunes — APPROVED** [review]

File: google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionOptions.java (hunk 1 above)
Comment (olavloite) [reply google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionOptions.java]: Done

File: google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionOptions.java (hunk 2 above)
Comment (olavloite) [reply google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionOptions.java]: Done

**Pull request merged**